src/main/java/org/opensearch/knn/index/KNNVectorSimilarityFunction.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.knn.index;
+
+import org.apache.lucene.index.VectorSimilarityFunction;
+import org.opensearch.knn.plugin.script.KNNScoringUtil;
+
+/**
+ * Wrapper class of VectorSimilarityFunction to support more function than what Lucene provides
+ */
+public enum KNNVectorSimilarityFunction {
```

Comment:
We should think of a better way here rather than just having a thin wrapper. Should the space type or any other class present in KNN do the vector calculation rather than we telling the caller that they have to call `VectorSimilarityFunction`

Reply:
If we want to improve in a separate PR lets have a GH issue for this. This PR is already very big and brining a lot of new classes that needs to be thought through if we are just putting everything on future PRs then we might loose the context here.

Reply:
I am not making anything worse with this thin wrapper I believe. The caller calling `VectorSimilarityFunction` is what we have now.